### PR TITLE
Template for unused shape error

### PR DESCRIPTION
**Pull request by KathleenQ**
Test is included.
The template looks like:
Screen Shot 2021-02-10 at 3 22 32 pm
The codes have already been run through clang-format.

### CODE REVIEW

File: webapp/src/main/webapp/resources/template.soy [hunk 1]
```diff
@@ -103,4 +103,34 @@
...
+
+
+/**
+ * Renders a basic explanation of the validation notice.
+ * @param notices : List<[shapeId : string, csvRowNumber : number]>
+ */
+{template .unusedShape}
+{let $numNotices: length($notices)/}
+  <p class="error">Error - Unused shape!</p>
+  <p>Description: Every shape in shapes.txt is used by some trip from trips.txt.</p>
```
Comment (jupi3200) [inline webapp/src/main/webapp/resources/template.soy]: should be used?

Comment (KathleenQ) [reply webapp/src/main/webapp/resources/template.soy]: Thank you for pointing it out! I have changed the description to "The shape in shapes.txt is never used by any trip from trips.txt.".

**Comment by KathleenQ** [comment]
All merged functions are all run through clang auto-formatting as well.

File: webapp/src/main/webapp/test/output.test.js [hunk 2]
```diff
@@ -107,27 +107,47 @@ describe('Output', function() {
...
+<thead>\
+<tr><th>Shape ID</th><th>CSV Row Number</th></tr>\
+</thead>\
+<tbody>\
+<tr><td>star2</td><td>13</td></tr>\
+</tbody>\
+</table>\
+<p>Please delete the unused shape(s)!</p><br><br></div>'
+    expect(document.getElementById('error').innerHTML).toContain(output);
+  });
```
Comment (KathleenQ) [inline webapp/src/main/webapp/test/output.test.js:131]: Only this section needs to review in this file.

**Review by jupi3200 — APPROVED** [review]

File: webapp/src/main/webapp/resources/output.js [hunk 3]
```diff
@@ -54,31 +54,43 @@ function invalid_row_length(params) {
...
+  const template = goog.soy.renderAsElement(
+      validator.templates.wrongParentLocationType, params);
+  document.getElementById('error').appendChild(template);
+}
+
+function unused_shape(params) {
+  const template =
+      goog.soy.renderAsElement(validator.templates.unusedShape, params);
+  document.getElementById('error').appendChild(template);
 }
```
Comment (KathleenQ) [inline webapp/src/main/webapp/resources/output.js:66]: Only this section needs to review in this file.

File: webapp/src/main/webapp/resources/template.soy [hunk 4]
```diff
@@ -103,6 +103,35 @@
...
+          <td>{$notice.shapeId}</td>
+          <td>{$notice.csvRowNumber}</td>
+        </tr>
+      {/foreach}
+    </tbody>
+  </table>
+  <p>Please delete the unused shape(s)!</p>
+  <br><br>
+{/template}
+
```
Comment (KathleenQ) [inline webapp/src/main/webapp/resources/template.soy:134]: Need to review.

**Review by KathleenQ — COMMENTED** [review]
All other sections are changes for auto-formatting after merging all PRs. No need to review.

**Pull request merged**